components/camel-google-bigquery/src/test/java/org/apache/camel/component/google/bigquery/unit/GoogleBigQueryProducerTest.java

```diff
@@ -0,0 +1,120 @@
+        assertEquals(1, requests.get(0).getRows().size());
+        assertNull(requests.get(0).getRows().get(0).getInsertId());
+    }
+
+
+    @Test
+    public void useAsInsertIdConfig() throws Exception {
+        configuration.setUseAsInsertId("row1");
+        Map<String, String> object = new HashMap<>();
+        object.put("row1", "value1");
+        producer.process(createExchangeWithBody(object));
+        ArgumentCaptor<TableDataInsertAllRequest> dataCaptor = ArgumentCaptor.forClass(TableDataInsertAllRequest.class);
+        verify(tabledataMock).insertAll(eq(projectId), eq(datasetId), eq(tableId), dataCaptor.capture());
+        List<TableDataInsertAllRequest> requests = dataCaptor.getAllValues();
+        assertEquals(1, requests.size());
+        assertEquals(1, requests.get(0).getRows().size());
+        assertEquals("value1", requests.get(0).getRows().get(0).getInsertId());
+    }
+
+    @Test
+    public void listOfMessages() throws Exception {
+        List<Map<String, String>> messages = new ArrayList<>();
+        messages.add(new HashMap<>());
+        messages.add(new HashMap<>());
+        producer.process(createExchangeWithBody(messages));
+        ArgumentCaptor<TableDataInsertAllRequest> dataCaptor = ArgumentCaptor.forClass(TableDataInsertAllRequest.class);
+        verify(tabledataMock).insertAll(eq(projectId), eq(datasetId), eq(tableId), dataCaptor.capture());
+        List<TableDataInsertAllRequest> requests = dataCaptor.getAllValues();
+        assertEquals(1, requests.size());
+        assertEquals(2, requests.get(0).getRows().size());
+    }
+
+    @Test
+    public void groupExchange() throws Exception {
+        Exchange exchange1 = createExchangeWithBody(new HashMap<>());
+        Exchange exchange2 = createExchangeWithBody(new HashMap<>());
+        Exchange exchange = createExchangeWithBody(null);
+        exchange.setProperty(Exchange.GROUPED_EXCHANGE, Lists.newArrayList(exchange1, exchange2));
+
+        producer.process(exchange);
+        ArgumentCaptor<TableDataInsertAllRequest> dataCaptor = ArgumentCaptor.forClass(TableDataInsertAllRequest.class);
+        verify(tabledataMock).insertAll(eq(projectId), eq(datasetId), eq(tableId), dataCaptor.capture());
+        List<TableDataInsertAllRequest> requests = dataCaptor.getAllValues();
+        assertEquals(1, requests.size());
+        assertEquals(2, requests.get(0).getRows().size());
+    }
+
+    @Test
+    public void groupExchangeDifferentTableSuffix() throws Exception {
+        Exchange exchange1 = createExchangeWithBody(new HashMap<>());
+        exchange1.getIn().setHeader(GoogleBigQueryConstants.TABLE_SUFFIX, "_SUFFIX1");
+        Exchange exchange2 = createExchangeWithBody(new HashMap<>());
+        exchange2.getIn().setHeader(GoogleBigQueryConstants.TABLE_SUFFIX, "_SUFFIX2");
+        Exchange exchange = createExchangeWithBody(null);
+        exchange.setProperty(Exchange.GROUPED_EXCHANGE, Lists.newArrayList(exchange1, exchange2));
```

Comment:
Avoid using Exchange.GROUPED_EXCHANGE as its not for components to use, its for the aggregate EIP pattern.